fabtests/functional/flood.c

```diff
@@ -0,0 +1,305 @@
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <unistd.h>
+
+#include <shared.h>
+
+int sleep_time = 0;
+
+static ssize_t post_one_tx(struct ft_context *msg)
+{
+	ssize_t ret;
+
+	if (ft_check_opts(FT_OPT_VERIFY_DATA | FT_OPT_ACTIVE)) {
+		ret = ft_fill_buf(msg->buf + ft_tx_prefix_size(),
+				  opts.transfer_size);
+		if (ret)
+			return ret;
+	}
+
+	return ft_post_tx_buf(ep, remote_fi_addr, opts.transfer_size,
+			      NO_CQ_DATA, &msg->context, msg->buf,
+			      msg->desc, 0);
+}
+
+static ssize_t wait_check_rx_bufs(void)
+{
+	ssize_t ret;
+	int i;
+
+	ret = ft_get_rx_comp(rx_seq);
+	if (ret)
+		return ret;
+
+	if (!ft_check_opts(FT_OPT_VERIFY_DATA | FT_OPT_ACTIVE))
+		return 0;
+
+	for (i = 0; i < opts.window_size; i++) {
+		ret = ft_check_buf((char *) rx_ctx_arr[i].buf +
+				   ft_rx_prefix_size(), opts.transfer_size);
+		if (ret)
+			return ret;
+	}
+
+	return 0;
+}
+
+static int post_rx_sync(void)
+{
+	int ret;
+
+	ret = ft_post_rx(ep, rx_size, &rx_ctx);
+	if (ret)
+		return ret;
+
+	if (opts.dst_addr) {
+		ret = ft_tx(ep, remote_fi_addr, 1, &tx_ctx);
+		if (ret)
+			return ret;
+
+		ret = ft_get_rx_comp(rx_seq);
+	} else {
+		ret = ft_get_rx_comp(rx_seq);
+		if (ret)
+			return ret;
+
+		ret = ft_tx(ep, remote_fi_addr, 1, &tx_ctx);
+	}
+
+	return ret;
+}
+
+static void mr_close_all(struct ft_context *ctx_arr, int window_size)
+{
+	int i;
+	for (i = 0; i < window_size; i++)
+		FT_CLOSE_FID(ctx_arr[i].mr);
+}
+
+static int run_seq_mr_send(void) {
+
+	int ret;
+	int i;
+
+	mr_close_all(tx_ctx_arr, opts.window_size);
+	mr_close_all(rx_ctx_arr, opts.window_size);
+
+	printf("Sequential memory registration:");
+	if (opts.dst_addr) {
+		for (i = 0; i < opts.window_size; i++) {
+			ret = ft_reg_mr(fi, tx_ctx_arr[i].buf, tx_mr_size,
+					ft_info_to_mr_access(fi),
+					FT_TX_MR_KEY + i, opts.iface, opts.device,
+					&(tx_ctx_arr[i].mr), &(tx_ctx_arr[i].desc));
+			if (ret)
+				goto err;
+
+			if (ft_check_opts(FT_OPT_VERIFY_DATA | FT_OPT_ACTIVE)) {
+				ret = ft_fill_buf(tx_ctx_arr[i].buf + ft_tx_prefix_size(),
+						  opts.transfer_size);
+			if (ret)
+				goto err;
+			}
+
+			ft_post_tx_buf(ep, remote_fi_addr,  opts.transfer_size,
```

Comment:
Can you replace this and the fill with the post_one_tx_buf function in this test?